examples/angular/src/app/guards/can-match-page.guard.ts

```diff
@@ -25,10 +25,28 @@ export const canMatchPage: CanMatchFn = async (
       entity: DotCMSPageAsset;
     };
 
+    const { vanityUrl } = entity;
+
+    if (vanityUrl?.permanentRedirect) {
+      router.navigate([vanityUrl.forwardTo]);
```

Comment:
@KevinDavilaDotCMS  a `CanMatchFn` should return a `boolean` or `UrlTree` for redirects cases therefore I suggest do redirect using a UrlTree , like this:

```ts
if (vanityUrl?.permanentRedirect) {
      return router.createUrlTree([vanityUrl.forwardTo]);
}
```

As a result, Angular can better handle the[ prioritization and cancellation route](https://juri.dev/blog/2018/11/better-route-guard-redirects/), note in v18 could use the RedirectCommand.